sympy/core/numbers.py

```diff
@@ -155,26 +155,32 @@ def igcd(*args):
     5
 
     """
-    a = args[0]
-    for b in args[1:]:
+    a = 0
+    args = list(args)
+    while args:
+        a = args.pop()
+        if a == 0:
+            continue
+        break
+    a = abs(as_int(a))
+    while args:
+        b = args.pop()
+        if b == 0:
+            continue
+        b = as_int(b)
```

Comment:
Although the sign change is handled below, handling it here would keep the cache from including negatives. I'll change it.